### ISSUE TITLE
Training a neural network in neuralnet has taken 3 consecutive days and not finished!

### ISSUE THREAD

**Issue by Maryam-Nasseri** [bips-hb/neuralnet#41]
I originally posted this as a question to the Stack Overflow community, but someone suggested that I raise this issue here, so I will copy the description here:

I have constructed a model in R (R Studio) for a neural network with two hidden layers using the neuralnet package. It is already 3 non-stop days but the model is not trained. All I see is the usual blinking cursor which is the sign of data in processing. There are no error or warning messages and nothing has crashed. All I need to know is if this is normal and how long I should expect before the model is trained and the object is created.

This is a relatively simple model with 1260 rows and 11 columns of continuous data as features predicting the 12th column ('labels') as a factor with six classes (multi-class classification task). Data has been properly scaled before splitting it to train and test datasets and the factor column of classes has been re-coded to numeric values with the revalue() in R. Here are the model specs:

 model = neuralnet(
        labels~A+B+C+D+E+F+G+H+I+J+K,        
    data=train_data,
    hidden=c(4,2),                
    linear.output = FALSE,       
    stepmax = 1e+06,             

    )
I have read many online posts including a number of Stack Overflow questions of similar nature but none gives me the answer I need. Is this caused by GPU or the complexity of the model? and how may I solve this issue? Here are my machine's specs:

Ubuntu 20.04.1 LTS 64 bit, RAM 48 GiB, processor AMD® Ryzen 5 2600 six-core processor × 12.

PS: I have used my own collected data so I'll not be able to share it, but I hope from the above description the maintainers could find the issue. Thanks.

**Comment by mnwright** [comment]
Hm... the package might not be the fastest neural network package around but I think it should manage a data/network like that. Could you try to simulate a similar dataset and share a reproducible example?

**Comment by Maryam-Nasseri** [comment]
Thanks, Marvin for the response. It turned out that I was using the merge() function incorrectly on my data preprocessing so it was adding a ton of new rows to my data. I corrected that and ran the model but the model with the same features still did not converge, even after increasing the stepmax and tuning the hidden layers. 

Just to note that the same model (on the same data) gave good results with nnet, and also the caret package with nnet method. I also tested it with Python in Keras and sci-kit learn (MLP in sklearn) and they both gave results. 

Here's the code on neuralnet package:

data<- read.csv("data.csv")

# revalue the classes column/character vector to numeric dummy codes
data <- 
  data %>%
  mutate(classes = revalue(classes, c("A" = 1, "B" = 2, "C" = 3, "D" = 4, "E" = 5, "F" = 6)))

# scaling numeric columns 
varnames<- c("classes")
index<- names(data) %in% varnames
data1<- scale(data[, !index])  
data1<- as.data.frame(scale(data[, !index]))
data1['labels']=data['labels']

# split training and testing datasets
train_idx <- sample(nrow(data), 0.7 * nrow(data))
train_data <- data[train_idx,]
test_data <- data[-train_idx,] 
data <- data %>% mutate_if(is.character, as.factor)

# first model which didn't converge:
model = neuralnet(
labels~a+b+c+d+e+f+g+h+i+j+k,   #11 numeric predictors
data=train_data,
hidden=c(4,2),
linear.output = FALSE,
stepmax = 1e+05
)

#second model which didn't converge:
model = neuralnet(
  abels~a+b+c+d+e+f+g+h+i+j+k, 
  data=train_data,
  hidden=c(6,2),             
  linear.output = FALSE,     
  stepmax = 1e+06,         
  err.fct = "ce",           
  lifesign= "minimal",
  act.fct = "logistic"      
)

Thanks again for the response.
